src/components/register.js

```diff
@@ -79,26 +93,71 @@ export default class RegisterComponent extends React.Component {
     );
   };
 
-  inputHandler = (event) => {
+  async getAvailableUsername(username) {
+    const client = new ApiClient(API_HOST);
+    return await client.getAvailableUsername(username);
+  }
+
+  async checkUserExists(username) {
+    const client = new ApiClient(API_HOST);
+    return await client.checkUserExists(username);
+  }
+
+  inputHandler = async (event) => {
+    if (this.usernameManuallyChanged)
+      return;
+
     const field = event.target;
-    const result = field.value.replace(/\W|\d/g, '');
+    const input = field.value.replace(/\W|\d/g, '');
 
     if (field.getAttribute('name') === 'firstName') {
-      this.first = result;
+      this.first = input;
     } else if (field.getAttribute('name') === 'lastName') {
-      this.last = result;
+      this.last = input;
+    }
+    
+    const result = this.first + this.last;
+    if (!result) {
+      this.setState({ username: result });
+      return;
     }
 
-    if (!this.usernameManuallyChanged) {
-      this.setState({ username: this.first + this.last });
+    try {
+      this.setState({ username: await this.getAvailableUsername(result) });
+      this.unavailable = false;
+      this.error = '';
+    } catch (e) {
+      this.error = e.message;
     }
   };
 
-  usernameInputHandler = (event) => {
+  usernameInputHandler = async (event) => {
     const result = event.target.value.replace(/\s|\W/g, '');
 
     this.setState({ username: result });
     this.usernameManuallyChanged = true;
+    
+    try {
+      this.unavailable = await this.checkUserExists(result);
+      this.error = '';
```

Comment:
so, `this.unavailable` might be set to `true` and `this.error` will be empty. this will lead to empty error message displayed?